src/runtime/dict.cpp

```diff
@@ -31,25 +31,57 @@ using namespace pyston::ExceptionStyle;
 using pyston::ExceptionStyle::ExceptionStyle;
 
 Box* dictRepr(BoxedDict* self) {
-    std::vector<char> chars;
-    chars.push_back('{');
-    bool first = true;
-    for (const auto& p : self->d) {
-        if (!first) {
-            chars.push_back(',');
+
+    try {
```

Comment:
Minor, but the try blocks should start after the `if (status != 0)` block is done -- ie we should only do it if the Py_ReprEnter call was hit and succeeded.  Technically the try block should also end right before the call to PyRepr_Leave in case that or boxString throw exceptions (though since those functions happen to not through exceptions it's less important).

Also, can we delete some of these new blank lines?